docs/metadata.md

```diff
@@ -0,0 +1,582 @@
+
+#### package.name
+
+The name of the package. Comes from the file that defines the package, such as
+`package.json`, `go.mod`, `setup.cfg` etc.
+
+#### package.version
+
+The version of the package. Usually comes from the file that defines the package, such
+as `package.json`, `setup.cfg` etc. Golang packages define their versions through git
+data, see https://golang.org/doc/modules/version-numbers.
+
+#### package.type
+
+The type of the package. Corresponds to the name of the package manager that Cachito was
+told to use when processing this package. The one exception is `go-package`, see
+[go-package type][go-package-type] in the README.
+
+#### package.path
+
+The relative path to the package directory from the root of the repository. Only present
+if the package is not in the root directory.
+
+#### package.dependencies
+
+The list of dependencies that Cachito downloaded for this package. See below.
+
+### dependencies
+
+The list of dependencies that Cachito downloaded. With some exceptions (see
+[dependency.version](#dependencyversion)), these do not come from the source repository.
+Dependencies appear both in `packages[].dependencies` and the top-level `dependencies`.
+The former is a per-package listing of dependencies while the latter is a merged and
+de-duplicated list of all the dependencies across all the packages.
+
+```json
+{
+  "name": "yarn",
+  "version": "1.22.10",
+  "type": "npm",
+  "dev": true,
+  "replaces": null
+}
+```
+
+In this somewhat funny example, Cachito downloaded `yarn` as a dev dependency of an
+`npm` package ([bootstrap-yarn](#packages)).
+
+#### dependency.name
+
+The name of the dependency. Comes from the file that defines what dependencies the
+package needs, such as `package-lock.json`, `go.mod`, `requirements.txt` etc.
+
+#### dependency.version
+
+The version of the dependency. Like the name, this comes from the file that defines the
+dependencies.
+
+Unlike the package version, which is normally a valid semantic version according to the
+package manager, the dependency versions have a much wider range of possible values.
+This is because most package managers let you specify dependencies not only with the
+name and version, but also support git repos, http(s) tarballs, local files etc. See
+[External Dependencies][feature-support] in the README.
+
+<details>
+<summary>Example - NPM dependency from a URL</summary>
+
+```json
+{
+  "name": "fecha",
+  "version": "https://github.com/taylorhakes/fecha/archive/91680e4db1415fea33eac878cfd889c80a7b55c7.tar.gz",
+  "type": "npm",
+  "dev": false,
+  "replaces": null
+}
+```
+</details>
+
+While the version value may not always be a valid semantic version, it *should* always
+be a valid way to specify the dependency within the rules of the given package manager.
+
+Now comes the time to explain why the part about dependencies never coming from the
+source repo was a lie. Package managers that support local files as dependencies can, in
```

Comment:
Where is the lie? Two paragraphs above it states that `local files` are allowed.

Reply:
Oh, true. I got lost in my own doc :sweat_smile:

Will rephrase

Reply:
Rephrased